### PR TITLE
Adding two new arctic shrub PFTs, and some arctic grass parameter updates

### PR DESCRIPTION
**Pull request by jenniferholm**
### Description:
To help improve the representation of arctic traits we are introducing two new shrubs that have been parameterized with an initial set of traits that more closely matches arctic observations, but improvements could still be made.  We have also updated a subset of parameter traits for the default c3 arctic grass PFT. 

These updates are based on the work done by Yanlan Liu and her 2024 parameter sensitive manuscript "Large divergence of projected high latitude vegetation composition and productivity due to functional trait uncertainty" https://agupubs.onlinelibrary.wiley.com/doi/full/10.1029/2024EF004563
We introduce the new "broadleaf_evergreen_arctic_shrub" and "broadleaf_colddecid_arctic_shrub" PFTs, and update the proportional breakdown of only the "broadleaf deciduous boreal shrub" in the hlm_pft_map. 
This is so that non-arctic shrubs can still exist and compete with arctic shrubs, and to distinguish the traits/characteristics between the arctic environments that are pretty distinct.

This PR addresses the discussion in Issue #1204 

### Collaborators:
Help from @rgknox, @YanlanLiu, and conversations with @ckoven, and others. 

### Expectation of Answer Changes:
Will change the results of shrubs in high latitudes. 

### Checklist

*If this is your first time contributing, please read the [**CONTRIBUTING**](https://github.com/NGEET/fates/blob/main/CONTRIBUTING.md) document.*

All checklist items must be checked to enable merging this pull request:

*Contributor*
- [x] The in-code documentation has been updated with descriptive comments
- [x] The documentation has been assessed to determine if updates are necessary

*Integrator*
- [x] FATES PASS/FAIL regression tests were run
- [x] Evaluation of test results for answer changes was performed and results provided

### Documentation

- [Technical Note](https://github.com/NGEET/fates-docs) update:
- [User's Guide](https://github.com/NGEET/fates-users-guide) update: 

### Test Results:

*CTSM (or) E3SM (specify which) test hash-tag:*

*CTSM (or) E3SM (specify which) baseline hash-tag:*

*FATES baseline hash-tag:*

*Test Output:*

### CODE REVIEW

**Comment by glemieux** [comment]
@jenniferholm and @rgknox to coordinate on running a long-term f45 smoke test.  If the results look good we'll go ahead and update the default parameter file with this instead of having a second param file.

**Comment by rgknox** [comment]
@jenniferholm and I are running some long-term smoke tests.  I generated the following error in a fixed-biogeography & no-comp type simulation:

```
The distribution of this host land model PFT :          10
 into FATES PFTs, does not add up to 1.0.
 Error is:  -1.00000000000000
 and the hlm_pft_map is:  0.000000000000000E+000  0.000000000000000E+000
  0.000000000000000E+000  0.000000000000000E+000  0.000000000000000E+000
  0.000000000000000E+000  0.000000000000000E+000  0.000000000000000E+000
  0.000000000000000E+000  0.000000000000000E+000  0.000000000000000E+000
  0.000000000000000E+000  0.000000000000000E+000  0.000000000000000E+000
 Aborting
 ENDRUN:
 ERROR in EDPftvarcon.F90 at line 2244
```

This means that the 10th pft from the ELM set, has nothing to map to in the FATES set.  This is a problem because we have areas in the surface file that will have no patches to assign to in FATES.

**Comment by rgknox** [comment]
Note that previously, this ELM PFT was mapping to:    ` 9 broadleaf_colddecid_extratrop_shrub  `

**Comment by rgknox** [comment]
@jenniferholm, I  see that the CDL I've generated is inconsistent with the hlm_pft_map you specified in the XML file. So this is either a problem with the batch-pathch script, or I did something wrong while creating the new cdl file.

File: parameter_files/archive/new_shrubPFT_patch_params.xml [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+      <fates_frag_seed_decay_rate> 0.35 </fates_frag_seed_decay_rate>
+      <fates_recruit_seed_germination_rate> 0.29 </fates_recruit_seed_germination_rate>
+    </pft_group>
+
+    <!-- Update the HLM-PFT map since PFTs 9, 10, 11 now share area -->
+    <!-- Splits the HLM deciduous boreal shrub (which must sum to 1.0)  -->
+    <!-- into 0.1 for the evergreen arctic, and 0.8 for cold-decid arctic -->
+    <!-- and leaves 0.1 in the orginial non-arctic extratrop cold-decid shrub -->
+    <pft_group ids="9" >
+      <fates_hlm_pft_map> 0, 0, 0, 0, 0, 0, 0, 0, 0, 0.0, 0.1, 0, 0, 0 </fates_hlm_pft_map>
```
Comment (rgknox) [inline parameter_files/archive/new_shrubPFT_patch_params.xml:122]: here's the problem, we zerod out the 10'th index in this vector. It should have stayed as 1.0

Comment (jenniferholm) [reply parameter_files/archive/new_shrubPFT_patch_params.xml:122]: @rgknox - Yep, you're exactly right. The 10th index in this vector should have remained as 1.0 (as it always has been in previous versions).  I was only thinking about ELM index 11 changes, and forgot to keep this as is.

**Comment by jenniferholm** [comment]
> @jenniferholm and I are running some long-term smoke tests. I generated the following error in a fixed-biogeography & no-comp type simulation:
> 
> ```
> The distribution of this host land model PFT :          10
>  into FATES PFTs, does not add up to 1.0.
>  Error is:  -1.00000000000000
>  and the hlm_pft_map is:  0.000000000000000E+000  0.000000000000000E+000
>   0.000000000000000E+000  0.000000000000000E+000  0.000000000000000E+000
>   0.000000000000000E+000  0.000000000000000E+000  0.000000000000000E+000
>   0.000000000000000E+000  0.000000000000000E+000  0.000000000000000E+000
>   0.000000000000000E+000  0.000000000000000E+000  0.000000000000000E+000
>  Aborting
>  ENDRUN:
>  ERROR in EDPftvarcon.F90 at line 2244
> ```
> 
> This means that the 10th pft from the ELM set, has nothing to map to in the FATES set. This is a problem because we have areas in the surface file that will have no patches to assign to in FATES.

Oops!  Oversight on my part that the ELM 10th PFT didn't add up to 1.0. Thanks for finding this

**Comment by rgknox** [comment]
The 40 year f45 smoke test with fixed biogeography and no-competition completed on perlmutter. Output here:
/global/cfs/cdirs/m2420/rgknox
I'll update later with some simple analysis.  Note the run was purely a smoke test to see if this created any model crashes, this is not to test the fidelity of the new parameterizations.

**Comment by glemieux** [comment]
Per fates software meeting today: @jenniferholm and @rgknox to coordinate on assessing long-duration smoke test results.

**Comment by rgknox** [comment]
A 40 year smoke test was ran.  Settings were pretty generic, but used fixed biogeography, no competition, and the new PFT set.  The following analysis reports gridded mean values on the 40th and last year of simulation:

https://drive.google.com/file/d/1YkwGd9gUz9zoc4C8AwpFl3LBM-l1xrdT/view?usp=sharing

**Comment by rgknox** [comment]
There isn't much coverage of PFTS 3,8 and 10.  The broadleaf arctic shrub 10 has a very small share of the total leaf mass.  And those needleleaf deciduous (larches?) are only in a few cells in siberia.  Also, we are still not mapping any ELM surface file classes to PFT 8 (broadleaf_hydrodecid_extratrop_shrub).  Should we use this opportunity to remove PFT 8?

**Comment by jenniferholm** [comment]
Hi @rgknox - related to your PFT 8 comment, I think we should discuss that in another issue and PR.  I tried to bring it up before, not it's not really related to the topic here, which is just the two arctic shrubs with more arctic traits. 

For the global sanity check, I think everything looks oaky with the two new PFTs. At least nothing crazy going on.  With the now PFT #10, we already knew that would have low coverage, and low leaf mass.  So that's expected.   I just wanted to make sure the new PFT #11 didn't take over too much (or out compete other cold deciduous plants), and I think it seems okay

**Comment by glemieux** [comment]
Apologies if I missed this discussion earlier.  I noticed in comparing the patch file update in #1136, which includes the grass allometry updates from @XiulinGao #1206, that the arctic shrub patch file has two values that conflict for the c3 arctic grass (PFT 10 in #1136 and PFT 12 in #1236).    

### Conflict 1: `fates_leaf_slatop`
PFT 12 (was PFT 10) from this PR updated to `0.027`:
https://github.com/NGEET/fates/blob/483905b1dbeec00d097f0d7dafd80db7746a5400/parameter_files/archive/new_shrubPFT_patch_params.xml#L108

PFT 10 from #1136 (now PFT 12) showing `0.05`:
https://github.com/NGEET/fates/blob/28b761574265a01b4c29dac931d5200bd896a70f/parameter_files/archive/api36.1.0_patch_params.xml#L87-L88

### Conflict 2: `fates_recruit_height_min`:
PFT 12 (was PFT 10) from this PR update to `0.11`:
https://github.com/NGEET/fates/blob/483905b1dbeec00d097f0d7dafd80db7746a5400/parameter_files/archive/new_shrubPFT_patch_params.xml#L106
PFT 10 from #1136 (now PFT 12) showing `0.2`:
https://github.com/NGEET/fates/blob/28b761574265a01b4c29dac931d5200bd896a70f/parameter_files/archive/api36.1.0_patch_params.xml#L90-L91

My assumption is that we want to use the updates from @YanlanLiu publication, but I wanted to double check.  
I wasn't entirely sure @jenniferholm, if you comment made in https://github.com/NGEET/fates/issues/1204#issuecomment-2221717981 was in reference to this:
> @XiulinGao - Thanks for your insights. In our work we still used the original allometry functions and modes and only updated the coefficients from our sensitivity analysis. It's great that you have functional forms that are more grass specific, and grass specific data. So I think the best plan is to stick with your PR https://github.com/NGEET/fates/pull/1206 (which has the improvements of grass specific allometry), for now, and then we can make updates later.

**Comment by jenniferholm** [comment]
Hi Greg, thanks for double checking these values.  Your assumption is correct that we are further updating _a few_ of the arctic c3 grass parameter beyond what was done in #1136, since we have values that are better suited for Arctic environments instead of Mediterranean grasses.  For example, the fates_recruit_height_min value of 0.2 from #1136 is too high for arctic grasses.  So it should be updated to 0.11 based on Yanlan's work.

**Comment by glemieux** [comment]
Thanks for the clarification @jenniferholm.

**Review by glemieux — APPROVED** [review]

**Comment by XiulinGao** [comment]
@glemieux @jenniferholm yeah look great to me! I think it totally makes sense to change the parameter when there is much better data for arctic grasses!

**Comment by glemieux** [comment]
Regression testing with just these updates to the parameter file results in expected b4b differences across multiple fates tests.  Spot checking the global SP test shows that the differences are located in the arctic as expected:

image

There was one failure due to ESCOMP/ctsm#2783; this is currently being retested on its own.  

Location: `/glade/u/home/glemieux/scratch/ctsm-tests/tests_pr1236-ctsm5029`

**Pull request merged**